### DISCUSSION TITLE
Trainable = False for layers/embedding layer

### DISCUSSION THREAD

**Discussion by sravanbabuiitm** [dmlc/gluon-nlp#333] [Q&A] [answered]
Hi,

While training a deep network with pretrained embeddings, LSTM , one would have a choice to freeze the pretrained embeddings or let the pretrained embeddings get trained too.

Other than not setting a trainer for the embeddings layer in which case it wont get updated, is there a diff way to explicitly mention while creating the layer to not train/update it ?

I need to create a trainer for rest of the network, which I m trying to avoid.

**Comment by szha** [comment, marked as answer]
you can do `net.layer1.collect_params().setattr('grad_req', 'null')` to freeze the weights in layer1.
